src/features/events/components/EventParticipantsTab/AddPersonButton.tsx

```diff
@@ -0,0 +1,138 @@
+import PersonAddIcon from '@mui/icons-material/PersonAdd';
+import { Box, IconButton, Popover } from '@mui/material';
+import { EmojiPeople, People } from '@mui/icons-material';
+import { useRef, useState } from 'react';
+
+import EventDataModel from 'features/events/models/EventDataModel';
+import messageIds from '../../l10n/messageIds';
+import ZUIFutures from 'zui/ZUIFutures';
+import { MUIOnlyPersonSelect as ZUIPersonSelect } from 'zui/ZUIPersonSelect';
+import { Msg, useMessages } from 'core/i18n';
+
+interface AddPersonButtonProps {
+  model: EventDataModel;
+}
+
+const AddPersonButton = ({ model }: AddPersonButtonProps) => {
+  const [anchorEl, setAnchorEl] = useState<Element | null>(null);
+
+  const messages = useMessages(messageIds);
+
+  const handleSelectedPerson = (personId: number) => {
+    model.addParticipant(personId);
+  };
+
+  const selectInputRef = useRef<HTMLInputElement>();
+
+  return (
+    <>
+      <IconButton
+        onClick={(ev) => {
+          setAnchorEl(ev.target as Element);
+        }}
+        sx={{ fontSize: '1rem' }}
+      >
+        <PersonAddIcon sx={{ mr: 1 }} />
+        <Msg id={messageIds.addPerson.addButton} />
+      </IconButton>
+
+      <Popover
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          horizontal: 'right',
+          vertical: 'bottom',
+        }}
+        elevation={1}
+        onClose={() => setAnchorEl(null)}
+        open={!!anchorEl}
+        PaperProps={{
+          style: {
+            display: 'flex',
+            flexDirection: 'column',
+            maxHeight: '40vh',
+            maxWidth: '400px',
+            width: '40vh',
+          },
+        }}
+        transformOrigin={{
+          horizontal: 'right',
+          vertical: 'top',
+        }}
+      >
+        <Box mt={1} p={2}>
+          <ZUIFutures
+            futures={{
+              participants: model.getParticipants(),
+              respondents: model.getRespondents(),
+            }}
+          >
+            {({ data: { participants, respondents } }) => {
+              const settingStatus = (personId: number) => {
```

Comment:
I think the name of this function is not very descriptive. Could it maybe be called `getOptionExtraLabel()` to reflect how it's used?